### PR TITLE
Add deep search in Request body

### PR DESCRIPTION
**Pull request by poadoc**
#322

### CODE REVIEW

**Comment by poadoc** [comment]
PR will be updated.

**Comment by poadoc** [comment]
Workability and performance have been tested with the attached OpenAPI definition file (size: ~5MB, number of methods: 900+)
[openapi_v3.zip](https://github.com/mrin9/RapiDoc/files/5377980/openapi_v3.zip)

File: src/utils/common-utils.js [hunk 1]
```diff
@@ -53,6 +53,27 @@ export function pathIsInSearch(searchVal, path) {
...
+  // We should use a separated string variable in order to exclude next error 'Uncaught RangeError: Invalid string length'
+  let recursiveStringToSearch = '';
+  stringToSearch = `${stringToSearch} ${Object.keys(properties || '').join(' ')}`;
+
+  Object.keys(properties).forEach((propertyName) => {
+    if (properties[propertyName].properties) {
+      recursiveStringToSearch = `${recursiveStringToSearch} ${processRequestBodyProperties(properties[propertyName].properties, stringToSearch)}`;
+    }
+    if (properties[propertyName].type === 'array') {
+      recursiveStringToSearch = `${recursiveStringToSearch} ${processRequestBodyProperties(properties[propertyName].items.properties, stringToSearch)}`;
```
Comment (mrin9) [inline src/utils/common-utils.js:70]: It seems like you are joining the string twice. the function should take an empty string initially and join it with `recursiveStringToSearch`


> I may be incorrect, just looking on a phone

**Review by mrin9 — CHANGES_REQUESTED** [review]

File: src/utils/common-utils.js [hunk 2]
```diff
@@ -53,6 +53,27 @@ export function pathIsInSearch(searchVal, path) {
   return stringToSearch.includes(searchVal);
 }
 
+function processRequestBodyProperties(properties, stringToSearch) {
```
[the remark this answers is not in the document]
Comment (poadoc) [reply src/utils/common-utils.js]: Completely agree. It has been fixed.

File: src/utils/common-utils.js (hunk 1 above)
Comment (poadoc) [reply src/utils/common-utils.js]: if (obj.properties[propertyName].properties) {
      recursiveresultStr = `${recursiveresultStr} ${joinObjectKeys(obj.properties[propertyName], resultStr)}`;
    }
if (obj.properties[propertyName].type === 'array') {
      recursiveresultStr = `${recursiveresultStr} ${joinObjectKeys(obj.properties[propertyName].items, resultStr)}`;
}

We can have both executing of these "if" statements in the context of the one function call. Therefore we have to join string from the first 'if' statement to the second one.

**Review by mrin9 — APPROVED** [review]

**Comment by mrin9** [comment]
Thank you for your PR !!!

**Review by mrin9 — COMMENTED** [review]
LGTM

**Pull request merged**